### PR TITLE
use Trio specific names and update instructions in testflight.md 

### PR DESCRIPTION
**Pull request by marionbarker**
* Add reference to github organization method which simplifies building many apps
* Add additional information previously only found in the docs.
* Include the fourth Identifier

This PR is in response to Issue #294

### CODE REVIEW

**Review by MikePlante1 — APPROVED** [review]
LGTM, thanks! I think this one should be merged to `dev` before 1.0.0 is released.

Here's some screenshots of the preview:


Screenshot 2024-06-11 at 12 52 30 PM
Screenshot 2024-06-11 at 12 51 16 PM

**Comment by bjornoleh** [comment]
If you look at my up to date screenshots from yesterday in https://github.com/nightscout/Trio/issues/294, I am unsure if this part is accurate, as I am not seeing the `FreeAPS` names anywhere in my screenshots? Surely the build targets are still called FreeAPS, but I think it will only bring confusion to mention this here in the context of GH builds and what to look for on  the Apple developer portal:

 Repeat this step for each **NAME** with associated **IDENTIFIER**:
    * FreeAPS: `org.nightscout.TEAMID.trio`
    * FreeAPS watchkitapp: `org.nightscout.TEAMID.trio.watchkitapp`
    * FreeAPS watchkitapp watchkitextension: `org.nightscout.TEAMID.trio.watchkitapp.watchkitextension`

**Comment by bjornoleh** [comment]
Ah, right! I expected differences due to Xcode, just didn’t know exactly what to expect, I thought it was only the XC prefix. 

Perhaps we should include instructions to search for Trio if there are many identifiers listed? That really clears it up for most.

**Comment by marionbarker** [comment]
@bjornoleh Are you requesting a modification? 

Should I put in a table of what the XC version looks like under NAMES? Or will most people be looking at the docs.
Does LiveActivity not get an XC in front of it?

**Comment by bjornoleh** [comment]
> @bjornoleh Are you requesting a modification?
> 
> Should I put in a table of what the XC version looks like under NAMES? Or will most people be looking at the docs. Does LiveActivity not get an XC in front of it?

A table with the typical names for Xcode vs Fastlane build would be great. 

Unsure about LiveActivity, perhaps this was added from a GitHub build because this feature was added later? I would think so.

**Comment by marionbarker** [comment]
### Summary
Do some testing prior to building a table for Identifier Names (Descriptions) for Mac-Xcode vs GitHub Actions (Browser Build) method for Trio.

### Question:
Do we really need LiveActivity as an identifier for Browser Build?

Tests with Trio alpha, commit `82f2d01`.

I deleted all identifiers (except Trio itself) and renamed the Trio description to be XC bundle_id

When I built with Mac-Xcode - these are the only identifiers on my identifiers page - no sign of LiveActivityExtension:

trio-identifiers-from-mac-xcode

I then ran some actions with my fork of Trio-alpha :

* Add Identifiers (alpha) - success
   * The previously shown identifiers are there but so is LiveActivity
   * LiveActivity Identifier details has AppGroup row checked
   * Add LoopGroup to the 3 trio identifiers (not Live activity) 

trio-identifiers-from-add-identifiers

* Create Certificates (alpha) - success
* Build Trio (alpha) 
   * I will update this comment after the build action finishes (it is 11 minutes in right now)
   * then I plan to delete the LiveActivity identifier and try again

**Comment by marionbarker** [comment]
I got more information via PM with @dnzxy.
I propose a different solution.
* I suggest a change to fastlane/Fastfile (patch below) - we could add to this PR (with an edited name) or close this and create a new PR
   * remove the AppGroup from LiveActivity (not needed and is confusing)
   * change the "name" for all identifiers to begin with Trio (so Identifier rows are adjacent and unique)
      * this separates Trio from iAPS Identifier rows 
* Changes of this type in Fastfile require corresponding changes in fastlane/testflight.md, so should all be in one PR 
* The documentation would then be changed accordingly with a PR to trio-docs

proposed-trio-identifiers

Patch for Fastfile: 
[trio-specific-identifiers.patch](https://github.com/user-attachments/files/15888222/trio-specific-identifiers.patch)

**Comment by marionbarker** [comment]
### Testing of Proposed Name Change

I created a branch, trio-specific-identifiers, at loopandlearn/Trio with this modification wrt nightscout/Trio alpha:
* https://github.com/loopandlearn/Trio/commit/b6db9605d1d73f32b5b79386ce8fb469b78605ee

I am in the process of testing this single Fastfile change after deleting Identifiers, Certificates and Match-Secrets.

**Comment by marionbarker** [comment]
If this modification is added to Trio, then Issue #174 can be closed.

**Comment by marionbarker** [comment]
Test successful with proposed change (see graphic).
(Forgot to mark on graphic) Delete all Trio XXX Identifiers
1. Clear previous certs: delete Match-Secrets and revoke distribution certificates
2. Successful Action: Add Identifiers
3. Add Trio App Group to Trio Watch and Trio WatchKit Extension
4. Successful Action: Create Certificates
5. Successful Action: Build Trio


Open questions:
* Consensus on desired names
* Decision whether to add Trio App Group to LiveActivity in case it is needed in the future

trio-specific-identifiers-test

**Comment by marionbarker** [comment]
This is ready for review.
* I confirmed that the LiveActivity Identifier does not require a shared app group
* Please test the instructions in testflight.md using the modified Fastfile in this PR
* For testing, you can delete all except the Trio identifier and then just Add Identifiers and edit for the App Group
* I also deleted identifiers and did Mac-Xcode build to capture that version for Identifiers

File: fastlane/testflight.md [hunk 1]
```diff
@@ -72,15 +76,13 @@ Log into your GitHub account to create a personal access token; this is one of t
...
 
-The first time you build with the GitHub Browser Build method for any DIY app, you will make up a password and record it as `MATCH_PASSWORD`. Note, if you later lose `MATCH_PASSWORD`, you will need to delete and make a new Match-Secrets repository (next step).
+The first time you build with the GitHub Browser Build method for any DIY app, you will make up a password and record it as `MATCH_PASSWORD`. Note, if you later lose `MATCH_PASSWORD`, you will need to delete and make a new Match-Secrets repository (next step). You use the same password for all DIY apps.
 
 ## Setup GitHub Match-Secrets Repository
 
-The creation of the Match-Secrets repository is a common step for all GitHub Browser Builds; do this step only once. You must be logged into your GitHub account.
-
-1. Create a [new empty repository](https://github.com/new) titled `Match-Secrets`. It should be private.
+> This step is no longer required, your Match-Secrets repository is automatically created the first time you run a GitHub Action. You can skip ahead to [Setup Github Trio repository](#setup-github-trio-repository).
```
Comment (bjornoleh) [inline fastlane/testflight.md:83]: Perhaps just remove this `## Setup GitHub Match-Secrets Repository` section then? Or write that it is created automatically.

**Review by bjornoleh — DISMISSED** [review]
Looks good and works well! The only thing I did not test was the optional renaming.

File: fastlane/testflight.md (hunk 1 above)
Comment (marionbarker) [reply fastlane/testflight.md]: I modified testflight.md and fixed this, added a few links and fixed a few typos.

**Review by dnzxy — APPROVED** [review]
Approving and merging.

**Pull request merged**